src/resizable.rs

```diff
@@ -87,6 +94,38 @@ impl<T> Queue<T> {
     pub fn available(&self) -> isize {
         self.queue.available()
     }
+    /// Initialize the notify_full sender
+    fn new_notify_full() -> FullNotifier {
+        let (sender, _) = watch::channel(());
+        sender
+    }
+    /// Check if the queue is full and notify any waiters
+    fn notify_if_full(&self) {
+        if self.push_semaphore.available_permits() == 0 {
+            self.notify_full.send_replace(());
+        }
+    }
+    /// Await until the queue is full.
+    pub async fn full(&self) {
+        if self.len() == self.capacity() {
+            return;
+        }
+        self.notify_full.subscribe().changed().await.unwrap();
+    }
+    /// Await until the queue is empty.
+    pub async fn empty(&self) {
+        loop {
+            let capacity = self.capacity().try_into().unwrap();
+            let _permit = self.push_semaphore.acquire_many(capacity).await.unwrap();
```

Comment:
Just like in the `limited` implementation I don't think this is correct. I guess the only (and correct) way around it is to use a channel for both `full` and `empty` notifications.